Add EOSIO

DanailMinchev opened the pull request:
**https://github.com/DanailMinchev/awesome-eosio**

**EOSIO is the most powerful infrastructure for decentralized applications.**


### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with the below requirements.


## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of sub-lists, preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

styfle commented on readme.md:
```diff
@@ -487,12 +487,15 @@
 - [Ripple](https://github.com/vhpoet/awesome-ripple#readme) - Open source distributed settlement network.
 - [Non-Financial Blockchain](https://github.com/machinomy/awesome-non-financial-blockchain#readme) - Non-financial blockchain applications.
 - [Mastodon](https://github.com/tleb/awesome-mastodon#readme) - Open source decentralized microblogging network.
+- [EOS](https://github.com/DanailMinchev/awesome-eos#readme) - EOS.IO is a decentralized operating system supporting industrial-scale applications.
```
Also change this PR title to `Add EOSIO` and no need to indicate which section

styfle replied:
A lot of sections have parenthesis such as `(developers)` but I don't think is this is necessary; it adds clutter.

styfle replied:
Many of the descriptions are missing or duplicates such as "Block explorer". Can these be more descriptive so it's clear how each entry in the list is different?

styfle replied:
The logo must be either fullwidth or placed at the top-right of the readme. It should also link to the EOSIO website.

DanailMinchev commented on readme.md:
```diff
@@ -487,12 +487,15 @@
 - [Network Analysis](https://github.com/briatte/awesome-network-analysis#readme)
 - [PCAPTools](https://github.com/caesar0301/awesome-pcaptools#readme)
 
+
```
I have added this line, because the sections ends with two lines and looks like we are missing one for `Decentralized Systems`.
For example all of the sections have two lines above their section names, please check:
- Networking
- Work
- Business
- etc

However, I have removed it - let me know if you want me to add it again and agree with the correction.

DanailMinchev commented:
Thank you for the review @styfle ! :)
I addressed your comments, could you review again?

styfle commented:
I think you can remove the "More information" lines. They are adding an unnecessary indentation level.

styfle commented:
Also, I recently found out that each description is supposed to end with a period. https://twitter.com/styfle/status/1025843172839055362

DanailMinchev commented:
@styfle all done, thanks

DanailMinchev commented:
Hi @sindresorhus , great project !

Updates about my PR: it was reviewed and approved ( thanks @styfle )

- As per your suggestions in https://github.com/sindresorhus/awesome/issues/1363, I have reviewed and commented on several PR's https://github.com/sindresorhus/awesome/pulls?utf8=%E2%9C%93&q=is%3Apr+reviewed-by%3ADanailMinchev to help the project.
- I've ran this through [awesome-lint](https://github.com/sindresorhus/awesome-lint) which pointed out some issues and I [fixed](https://github.com/DanailMinchev/awesome-eosio/commit/914434ee48f065da9d3b1a18355fe0d558bdebe9) them.
- [awesome-lint](https://github.com/sindresorhus/awesome-lint) was very useful and I put a comment with the full report in my PR reviews.

Here is the list of my reviews/comments:

**Reviews**

- https://github.com/sindresorhus/awesome/pull/1069
- https://github.com/sindresorhus/awesome/pull/1378
- https://github.com/sindresorhus/awesome/pull/1380

**Second review**

- https://github.com/sindresorhus/awesome/pull/1053#pullrequestreview-146265000

**Comments**

- https://github.com/sindresorhus/awesome/pull/1377#pullrequestreview-145859257
- https://github.com/sindresorhus/awesome/pull/1373#pullrequestreview-146258892

@sindresorhus could you let me know if everything is ok or should I address something else to move forward?

Thank you!

DanailMinchev commented:
@sindresorhus could you let me know if something else needs to be done?

sindresorhus commented:
```
❯ awesome-lint https://github.com/DanailMinchev/awesome-eosio
✖ Linting

  readme.md:192:3
  ✖  154:3  Link to https://www.buyeosram.io/ is dead                                                remark-lint:no-dead-urls
  ✖  192:3  Link to https://www.udemy.com/learning-blockchain-development-with-eos-and-cpp/ is dead  remark-lint:no-dead-urls
  ✖  193:3  Link to https://www.udemy.com/starting-with-eos/ is dead                                 remark-lint:no-dead-urls
  ✖  334:3  Link to https://eosprojects.org/ is dead                                                 remark-lint:no-dead-urls

  4 errors
```

https://github.com/sindresorhus/awesome-lint

sindresorhus commented:
Can you fix the merge conflict?

sindresorhus commented:
> Includes a project logo/illustration whenever possible.
> **The image should be high-DPI. Set it to maximum half the width of the original image.**

sindresorhus commented:
In the `Official` section, there's no need to prefix everything with `EOSIO`. That already implied and obvious.

sindresorhus commented:
Items in the `Blogs` and `YouTube Channels` section should have a description.

sindresorhus commented:
Looks weird to have a linebreak between these paragraphs:

```

EOSIO is the most powerful infrastructure for decentralized applications and open source smart contract platform.
EOSIO software enables businesses to rapidly build and deploy high-performance and high-security blockchain-based applications.
```

DanailMinchev commented:
Thank you for the review @sindresorhus !

I have addressed your comments, could you re-review please?

Some information:

- about the `awesome-lint` report:

```
❯ awesome-lint https://github.com/DanailMinchev/awesome-eosio
✖ Linting

  readme.md:192:3
  ✖  154:3  Link to https://www.buyeosram.io/ is dead                                                remark-lint:no-dead-urls
  ✖  192:3  Link to https://www.udemy.com/learning-blockchain-development-with-eos-and-cpp/ is dead  remark-lint:no-dead-urls
  ✖  193:3  Link to https://www.udemy.com/starting-with-eos/ is dead                                 remark-lint:no-dead-urls
  ✖  334:3  Link to https://eosprojects.org/ is dead                                                 remark-lint:no-dead-urls

  4 errors
```

Looks like only https://www.buyeosram.io/ is invalid.

Following links are *valid*:
https://www.udemy.com/learning-blockchain-development-with-eos-and-cpp/
https://www.udemy.com/starting-with-eos/
https://eosprojects.org/

I have removed them anyway, but would like to add them if you think it is `awesome-lint` bug, please let me know.

I have uploaded a new logo as well and addressed all other comments.

Cheers

sindresorhus commented:
> I have removed them anyway, but would like to add them if you think it is awesome-lint bug, please let me know.

It's probably this issue: https://github.com/transitive-bullshit/check-links/issues/4 Feel free to add them back.

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1102208276840640517
